tools/cli/adminCommands.go

```diff
@@ -183,6 +186,99 @@ func describeMutableState(c *cli.Context) *adminservice.DescribeMutableStateResp
 	return resp
 }
 
+// RegisterNamespace register a namespace
+func RegisterNamespace(c *cli.Context) error {
+	namespace := getRequiredGlobalOption(c, FlagNamespace)
+
+	description := c.String(FlagDescription)
+	ownerEmail := c.String(FlagOwnerEmail)
+
+	client := cFactory.FrontendClient(c)
+
+	var err error
+	retention := defaultNamespaceRetention
+	if c.IsSet(FlagRetention) {
+		retention, err = timestamp.ParseDurationDefaultDays(c.String(FlagRetention))
+		if err != nil {
+			return fmt.Errorf("option %s format is invalid: %s", FlagRetention, err)
+		}
+	}
+
+	var isGlobalNamespace bool
+	if c.IsSet(FlagIsGlobalNamespace) {
+		isGlobalNamespace, err = strconv.ParseBool(c.String(FlagIsGlobalNamespace))
+		if err != nil {
+			return fmt.Errorf("option %s format is invalid: %w.", FlagIsGlobalNamespace, err)
+		}
+	}
+
+	namespaceData := map[string]string{}
+	if c.IsSet(FlagNamespaceData) {
+		namespaceDataStr := c.String(FlagNamespaceData)
+		namespaceData, err = parseNamespaceDataKVs(namespaceDataStr)
+		if err != nil {
+			return fmt.Errorf("option %s format is invalid: %s", FlagNamespaceData, err)
+		}
+	}
+	if len(requiredNamespaceDataKeys) > 0 {
+		err = checkRequiredNamespaceDataKVs(namespaceData)
+		if err != nil {
+			return fmt.Errorf("namespace data missed required data: %s", err)
+		}
+	}
+
+	var activeClusterName string
+	if c.IsSet(FlagActiveClusterName) {
+		activeClusterName = c.String(FlagActiveClusterName)
+	}
+
+	var clusters []*replicationpb.ClusterReplicationConfig
+	if c.IsSet(FlagClusters) {
+		clusterStr := c.String(FlagClusters)
+		clusters = append(clusters, &replicationpb.ClusterReplicationConfig{
+			ClusterName: clusterStr,
+		})
+		for _, clusterStr := range c.Args() {
```

Comment:
what is c.Args()? could you verify that this part (line 236 to 246) works as expected?

Reply:
args are parameters passed after the flags. This is quite bad design, some details and the action we can take to improve the UX are here https://github.com/temporalio/temporal/pull/2176#discussion_r750724972